src/object/srv_ec_aggregate.c

```diff
@@ -2277,6 +2262,20 @@ agg_iterate_pre_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 		return rc;
 	}
 
+	agg_param->ap_credits++;
+	if (agg_param->ap_credits > agg_param->ap_credits_max) {
+		agg_param->ap_credits = 0;
+		*acts |= VOS_ITER_CB_YIELD;
+		D_DEBUG(DB_EPC, "EC aggregation yield type %d. acts %u\n",
+			type, *acts);
+		if (!(*acts & VOS_ITER_CB_SKIP))
+			agg_reset_pos(type, agg_entry);
+		if (ec_aggregate_yield(agg_param)) {
+			D_DEBUG(DB_EPC, "EC aggregation aborted\n");
+			rc = 1;
```

Comment:
just curious any difference to put this section of code to  agg_iterate_pre_cb() or agg_iterate_post_cb()?
seems the code is same (except you remove one log "D_DEBUG(DB_EPC, "EC aggregation yield type %d.\n", type);"?

and the commit msg "If there are partial stripes being processed, it should set SKIP, so the extent can be skipped after yield", seems it already has been set in the original code?